### DISCUSSION TITLE
posts/sqldatabase-active-directory-authent

### DISCUSSION THREAD

**Discussion by giscus** [TechWatching/techwatching.dev#36] [Announcements]
# posts/sqldatabase-active-directory-authent

https://www.techwatching.dev/posts/sqldatabase-active-directory-authent

**Comment by karpikpl** [comment]
I'm wondering if using dapper wouldn't be a better option. 
With `sqlcmd` the solution requires it to be installed and makes assumption about Interpreter, it needs to be `bash` on Linux/OSX and `pwsh` on Windows, which makes this solution platform specific.

**Reply by TechWatching** [reply]
Indeed it does make an assumption about the Interpreter but `pwsh` could be used for everything: Linux/OSX and Windows. 

Yes you are right, as I said in my article using `sqlcmd`  is probably no the best solution.
>Of the 3 possible solutions let's take the 3rd one with the Command provider and the sqlcmd utility. It is probably not the "best" solution but I thought it would be simpler to use the sqlcmd utility than writing a complete provider or even custom C# code to do the same. Furthermore, it's the opportunity to test the Command provider which is fairly new.

I think using Dapper with dynamic providers would be the best option, unfortunately .NET dynamic providers are not yet supported as I mention in the "Custom C# code". So you would have to use the `Apply` method. If you implement it, I would be interested in seeing how the code looks doing that!
